src/components/lateralBar/lateralBar.tsx

```diff
@@ -0,0 +1,157 @@
+import collapse from '../../../public/icons/collapse.svg'
+import type { Route } from '@/routes'
+import { appRoutes } from '@/routes'
+import { usePathname } from 'next/navigation'
+import classnames from 'classnames'
+import { FeedbackButton } from '../buttons/feedback/feedbackButton'
+import { useAppStore } from '@/stores/appStore'
+import type { LocalStorageState } from '@/hooks/useLocalStorage'
+import { useLocalStorage } from '@/hooks/useLocalStorage'
+
+const socialMedias = [
+  {
+    id: 'twitter',
+    href: 'https://twitter.com/OKP4_Protocol'
+  },
+  {
+    id: 'linkedin',
+    href: 'https://www.linkedin.com/company/okp4-open-knowledge-platform-for/'
+  },
+  {
+    id: 'discord',
+    href: 'https://discord.com/invite/okp4'
+  },
+  {
+    id: 'medium',
+    href: 'https://blog.okp4.network'
+  }
+]
+
+const groupedRoutes = appRoutes.reduce<Record<string, Omit<Route, 'group'>[]>>(
+  (acc, { name, id, path, group }) => ({
+    ...acc,
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    [group]: acc[group] ? [...acc[group], { name, id, path }] : [{ name, id, path }]
+  }),
+  {}
+)
+
+const localStorageKey = 'okp4-theme'
+
+// eslint-disable-next-line max-lines-per-function
+const LateralBar: FC = () => {
+  const isExpanded = useAppStore(store => store.isExpanded)
+  const expandLateralBar = useAppStore(store => store.expand)
+  const collapseLateralBar = useAppStore(store => store.collapse)
+  const { theme, setTheme }: ThemeContextType = useTheme()
+  const [, setValue]: LocalStorageState = useLocalStorage(localStorageKey)
+  const pathname = usePathname()
+
+  const handleCheck = useCallback(
+    (checked: boolean): void => {
+      const selectedTheme = checked ? 'light' : 'dark'
+      setTheme(selectedTheme)
+      setValue(selectedTheme)
+    },
+    [setTheme, setValue]
+  )
+  return (
+    <div
+      className={classnames('okp4-dataverse-portal-lateral-bar-main', {
+        collapsed: !isExpanded
+      })}
+    >
+      <div
+        className="okp4-dataverse-portal-lateral-bar-collapse-container"
+        onClick={isExpanded ? collapseLateralBar : expandLateralBar}
+      >
+        <Image alt="expand-lateral-bar-icon" src={isExpanded ? collapse : expand} />
+      </div>
+      <div
+        className={classnames('okp4-dataverse-portal-lateral-bar-title-container', {
+          collapsed: !isExpanded
+        })}
+      >
+        <Okp4Logo logoOnly={!isExpanded} />
+        <Switch
+          icons={{
+            checked: <Image alt="sun-icon" src={sunIcon} />,
+            notChecked: <Image alt="moon-icon" src={moonIcon} />
+          }}
+          isChecked={theme === 'light'}
+          onCheckedChange={handleCheck}
+        />
+      </div>
+      <div
+        className={classnames('okp4-dataverse-portal-lateral-bar-content-container', {
+          collapsed: !isExpanded
+        })}
+      >
+        {Object.entries(groupedRoutes).map((group, index) => (
```

Comment:
I don't know what you think, but maybe the code could have been split into smaller JSX elements to improve readability? It's a detail ;)